include/geode/mesh/core/detail/vertex_cycle.h

```diff
@@ -52,80 +52,147 @@ namespace geode
         }
 
         template < typename Container >
-        inline void rotate( Container& vertices )
+        inline void oriented_rotate( Container& vertices )
         {
             if( vertices.size() < 2 )
             {
                 return;
             }
             polygon_sort( vertices );
+        }
+
+        template <>
+        inline void oriented_rotate( std::array< index_t, 2 >& vertices )
+        {
+            if( vertices.front() > vertices.back() )
+            {
+                std::swap( vertices.front(), vertices.back() );
+            }
+        }
+
+        template < typename Container >
+        inline void remove_orientation( Container& vertices )
+        {
             if( vertices[1] > vertices.back() )
             {
                 std::reverse( vertices.begin() + 1, vertices.end() );
             }
         }
 
+        template < typename Container >
+        inline void rotate( Container& vertices )
+        {
+            oriented_rotate( vertices );
+            remove_orientation( vertices );
+        }
+
         template <>
         inline void rotate( std::array< index_t, 2 >& vertices )
```

Comment:
Is `rotate` still used?

Reply:
No, but wouldn't removing it be a breaking change ?